src/tdsops.f90

```diff
@@ -0,0 +1,700 @@
+
+      class(tdsops_t), intent(inout) :: self
+      character(*), intent(in) :: variation
+      character(*), optional, intent(in) :: bc_top, bc_bot
+      logical, optional, intent(in) :: paire
+
+      real(dp), allocatable :: dist_b(:)
+      real(dp) :: afi, bfi, dx
+      integer :: i, n, n_halo
+
+      dx = self%delta
+
+      ! alpha is alfa
+
+      select case(variation)
+      case('compact6')
+         self%alpha = 1._dp/3._dp
+         afi = 7._dp/9._dp/self%delta
+         bfi = 1._dp/36._dp/self%delta
+      case default
+         error stop 'variation is not defined'
+      end select
+
+      self%coeffs(:) = [0._dp, 0._dp, -bfi, -afi, &
+                        0._dp, &
+                        afi, bfi, 0._dp, 0._dp]
+
+      do i = 1, self%n_halo
+         self%coeffs_t(i,:) = self%coeffs(:)
+         self%coeffs_b(i,:) = self%coeffs(:)
+      end do
+
+      self%dist_sa(:) = self%alpha; self%dist_sc(:) = self%alpha
+
+      n = self%n
+      n_halo = self%n_halo
+
+      allocate(dist_b(n))
+      dist_b(:) = 1._dp
+
+      select case (bc_top)
+      case ('neumann')
+         if (present(paire) .and. paire) then
+            ! paire == 1; d(uu)/dx, dv/dx, dw/dx
+            !             d(vv)/dy, du/dy, dw/dy
+            !             d(ww)/dz, du/dz, dv/dz
+            self%dist_sa(1) = 0._dp
+            self%dist_sc(1) = 0._dp
+            self%coeffs_t(1, :) = [0._dp, 0._dp, 0._dp, 0._dp, &
+                                   0._dp, &
+                                   0._dp, 0._dp, 0._dp, 0._dp]
+            self%coeffs_t(2, :) = [0._dp, 0._dp, 0._dp, -afi, &
+                                   -bfi, &
+                                   afi, bfi, 0._dp, 0._dp]
+         else
+            ! paire == 0; d(uv)/dx, d(uw)/dx, du/dx
+            !             d(vu)/dy, d(vw)/dy, dv/dy
+            !             d(wu)/dz, d(wv)/dz, dw/dz
+            self%dist_sa(1) = 0._dp
+            self%dist_sc(1) = 2*self%alpha
+            self%coeffs_t(1, :) = [0._dp, 0._dp, 0._dp, 0._dp, &
+                                   0._dp, &
+                                   2*afi, 2*bfi, 0._dp, 0._dp]
+            self%coeffs_t(2, :) = [0._dp, 0._dp, 0._dp, -afi, &
+                                   bfi, &
+                                   afi, bfi, 0._dp, 0._dp]
+         end if
+      case ('dirichlet')
+         ! first line
+         self%dist_sa(1) = 0._dp
+         self%dist_sc(1) = 2._dp
+         self%coeffs_t(1, :) = [0._dp, 0._dp, 0._dp, 0._dp, &
+                                -2.5_dp/dx, &
+                                2._dp/dx, 0.5_dp/dx, 0._dp, 0._dp]
+         ! second line
+         self%dist_sa(2) = 0.25_dp
+         self%dist_sc(2) = 0.25_dp
+         self%coeffs_t(2, :) = [0._dp, 0._dp, 0._dp, -0.75_dp/dx, &
+                                0._dp, &
+                                0.75_dp/dx, 0._dp, 0._dp, 0._dp]
+      end select
+
+      select case (bc_bot)
+      case ('neumann')
```

Comment:
I think the name is misleading. In case of symmetry, a variable can be odd or even. In the absence of symmetry, one can use Dirichlet / Neumann bounday conditions. However, homogeneous Neumann is a very specific Neumann condition. From a generic point of view, I think we have the options :

1. Periodic
2. Symmetry (odd / even)
3. Dirichlet / Neumann with imposed value or flux

Reply:
We might want to skip this issue if the scope of the PR is limited to "mimick the current features of the current master branch of incompact3d" https://github.com/xcompact3d/x3d2/pull/11#discussion_r1368889905

However, I think the description of the boundary conditions above is sharper than the current one.